squad/frontend/views.py

```diff
@@ -225,16 +223,47 @@ def build(request, group_slug, project_slug, version):
     for status in __statuses__:
         test_results.add_status(status)
 
+    has_extra_metadata = False
+    if build.non_important_metadata:
+        has_extra_metadata = True
     context = {
         'project': project,
         'build': build,
         'test_results': test_results,
         'metadata': sorted(build.important_metadata.items()),
-        'extra_metadata': sorted(build.non_important_metadata.items()),
+        'has_extra_metadata': has_extra_metadata,
     }
     return render(request, 'squad/build.html', context)
 
 
+@auth
+def build_metadata(request, group_slug, project_slug, version):
+    group = Group.objects.get(slug=group_slug)
+    project = group.projects.get(slug=project_slug)
+
+    if version == 'latest-finished':
+        status = ProjectStatus.objects.prefetch_related('build').filter(
+            build__project=project,
+            finished=True,
+        ).order_by('build__datetime').last()
+        if status is None:
+            raise Http404()
+        build = status.build
+    else:
+        build = get_object_or_404(
+            project.builds.prefetch_related('test_runs'),
+            version=version,
+        )
```

Comment:
I don't think we need to support the "latest-finished" convention here, since we don't support it in any other view besides the main build page.